### ISSUE TITLE
List (s) give incorrect type FullName (s)

### ISSUE THREAD

**Issue by michaelcheers** [bridgedotnet/Bridge#1294]
[Bridge Fiddle](http://live.bridge.net/#f9543491757163d6696af0dca4795834)
[Dot Net Fiddle](https://dotnetfiddle.net/PYKzPK)
### Expected

System.Collections.Generic.List`1[[System.Int32, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]]
### Actual

Bridge.List$1$Bridge.Int32
### Steps To Reproduce

``` csharp
public class App
{
    [Ready]
    public static void Main()
    {
        List<int> value = new List<int>();
        Console.WriteLine(value.GetType().FullName);
    }
}
```

**Comment by geoffreymcgill** [comment]
Related to #1248, #1292, #1293 and #1184.

**Comment by LeonidVeriga** [comment]
Fixed as part of #778, #15 and #1337 (#1394).
Released with v15.0
